File: 25-reverse-nodes-in-k-group/solution.ml
exception Uneven
let rec rev_k_groups l k =
    try
        let (rest, acc) = go l k (fun x -> x) in
        acc (rev_k_groups rest k)
    with
    | Uneven -> l
and go l k acc =
    if k = 0 then (l, acc) else
        match l with
        | [] -> raise Uneven
        | x::xs -> go xs (k-1) (fun l -> x :: acc l)

let ex = [1;2;3;4;5]
let r = rev_k_groups ex 2

File: 25-reverse-nodes-in-k-group/solution.py
class Uneven(Exception):
    pass

class Solution:
    def reverseKGroup(self, head: Optional[ListNode], k: int) -> Optional[ListNode]:
        if k <= 1 or head is None:
            return head

        acc = p = head
        p = p.next
        acc.next = None
        try:
            for i in range (k-1):
                if p is None:
                    raise Uneven()
                else:
                    print(p.val)
                    t = p.next
                    p.next = acc
                    acc = p
                    p = t
        except Uneven:
            p = acc
            acc = None
            while p is not None:
                t = p.next
                p.next = acc
                acc = p
                p = t
            return acc
        else:
            head.next = self.reverseKGroup(p, k)
            return acc